### PR TITLE
glib: update to 2.68.0

### PR DESCRIPTION
**Pull request by paper42**
#### General
- [ ] This is a new package and it conforms to the [quality requirements](https://github.com/void-linux/void-packages/blob/master/Manual.md#quality-requirements)

#### Have the results of the proposed changes been tested?
- [x] I use the packages affected by the proposed changes on a regular basis and confirm this PR works for me
- [ ] I generally don't use the affected packages but briefly tested this PR

### CODE REVIEW

**Comment by ericonr** [comment]
This one introduces some relevant new APIs, so I would increase the `common/shlibs` requirement.

**Comment by ericonr** [comment]
If you want to make a separate PR for 2.66.8, we can merge that first. 2.68.0 is a bigger release.

**Comment by paper42** [comment]
I would prefer to work on this PR, but if someone else updates the package to 2.66.8, I can just rebase this. This update will be neccesary for Gnome 40.

**Comment by paper42** [comment]
I ran tests on my computer without patching them out and they all passed in a glibc and musl masterdir. I think I might be doing something wrong, because some of the tests are known to fail on musl.

File: srcpkgs/glib/patches/disable-broken-tests.patch [hunk 1]
```diff
@@ -20,33 +36,39 @@ index d8ebd56..be72f5c 100644
...
    'mutex' : {},
    'node' : {},
    'once' : {},
 -  'option-context' : {},
    'option-argv0' : {},
    'overflow' : {},
    'overflow-fallback' : {
+-- 
+2.31.0
+
```
Comment (Gottox) [inline srcpkgs/glib/patches/disable-broken-tests.patch:74]: A few of this tests can be fixed by adding `python3-pytest` to checkdepends.

**Comment by ericonr** [comment]
> I ran tests on my computer without patching them out and they all passed in a glibc and musl masterdir. I think I might be doing something wrong, because some of the tests are known to fail on musl.

I think so; how are you running them? I usually do `xbulk -Q glib`.

**Comment by ericonr** [comment]
I think we should include this patch: https://gitlab.gnome.org/GNOME/glib/-/merge_requests/2010 , it was a 2.68 regression.

**Comment by paper42** [comment]
@Gottox do you want to take over this PR?

**Comment by Gottox** [comment]
@paper42 Thanks, I cherry picked this commit. :+1:

**Pull request closed without merging**